### PR TITLE
LNX-193 Fix bug with agent delete

### PR DESCRIPTION
**Pull request by Krystian030**

### CODE REVIEW

File: src/routers/scene_router.py [hunk 1]
```diff
@@ -65,6 +66,10 @@ async def add(r: AddObjectDTO, background_tasks: BackgroundTasks):
 async def delete(object_id: int):
     logger.debug(f"Starting to delete object {object_id} from scene")
     state.scene.add_to_pending_actions(RemoveObject(object_id).serialize())
+    process_data: ProcessData = state.processes.pop(object_id, None)
```
Comment (filip-szweda) [inline src/routers/scene_router.py]: Move to `remove_process` helper function

File: src/routers/scene_router.py [hunk 2]
```diff
@@ -65,6 +66,10 @@ async def add(r: AddObjectDTO, background_tasks: BackgroundTasks):
...
+    if process_data is not None:
```
Comment (filip-szweda) [inline src/routers/scene_router.py]: `if process_data:`

File: src/routers/scene_router.py [hunk 3]
```diff
@@ -65,6 +66,10 @@ async def add(r: AddObjectDTO, background_tasks: BackgroundTasks):
...
+        process_data.process.terminate()
```
Comment (filip-szweda) [inline src/routers/scene_router.py]: Consider error handling

File: src/routers/scene_router.py [hunk 4]
```diff
@@ -65,6 +66,10 @@ async def add(r: AddObjectDTO, background_tasks: BackgroundTasks):
 async def delete(object_id: int):
     logger.debug(f"Starting to delete object {object_id} from scene")
     state.scene.add_to_pending_actions(RemoveObject(object_id).serialize())
```
Comment (filip-szweda) [inline src/routers/scene_router.py:69]: Check if the `Object` exists in the map first

**Review by filip-szweda — CHANGES_REQUESTED** [review]

**Review by EasyJB — APPROVED** [review]

File: src/routers/scene_router.py (hunk 3 above)
Comment (Krystian030) [reply src/routers/scene_router.py]: Terminate doesn't raise an excpeption upon invocation.

File: src/utils/helper_functions.py [hunk 5]
```diff
@@ -150,3 +150,15 @@ async def calculate_player_rankings():
 
     ranking.sort(key=lambda x: x[1], reverse=True)
     return ranking
+
+
+def terminate_process_for_object(object_id: int):
+    logger.debug(f"Terminating process for object {object_id}")
```
Comment (filip-szweda) [inline src/utils/helper_functions.py]:
```suggestion
    logger.debug(f"Terminating process for object with id: {object_id}")
```

File: src/utils/helper_functions.py [hunk 6]
```diff
@@ -150,3 +150,15 @@ async def calculate_player_rankings():
 
     ranking.sort(key=lambda x: x[1], reverse=True)
     return ranking
+
+
+def terminate_process_for_object(object_id: int):
```
Comment (filip-szweda) [inline src/utils/helper_functions.py]:
```suggestion
def terminate_process_for_object(object_id: int) -> None:
```

File: src/utils/helper_functions.py [hunk 7]
```diff
@@ -150,3 +150,15 @@ async def calculate_player_rankings():
...
+    logger.debug(f"Terminating process for object {object_id}")
+    if not state.scene.get_object_by_id(object_id):
+        logger.debug(f"Object {object_id} doesn't exist on the scene, skipping termination")
+        return
+
+    process_data = state.processes.pop(object_id, None)
+    if process_data:
+        process_data.process.terminate()
+    logger.debug(f"Process for object {object_id} has been successfully terminated")
```
Comment (filip-szweda) [inline src/utils/helper_functions.py]:
```suggestion
    logger.debug(f"Process has been successfully terminated for object with id: {object_id} ")
```

**Review by filip-szweda — APPROVED** [review]
I've added a few suggestions, but otherwise the change is ready to go :)

**Pull request merged**